hashicorp/dev-portal
# feat: add heap-track to top nav

## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds `data-heap-track` attributes to elements in the top navigation bar.

## 🤷 Why

To ensure elements in the top navigation bar can be consistently tracked in Heap.

## 🛠️ How

- adds data-heap-track to dropdown-menu
- adds data-heap-track to primary-nav-link
- adds data-heap-track to give-feedback-button (via `ButtonLink`)
- adds data-heap-track to top-nav logo link
- adds data-heap-track to top nav product-logo-link

## 🧪 Testing

- Visit the [home page][preview], and confirm attributes appear as expected
    - The logo link should have a `data-heap-track="navigation-header-home-logo-link"` attribute
    - The products dropdown activator should have a `data-heap-track="navigation-header-dropdown-menu-activator"` attribute on the `<button />` element
    - Each item in the products dropdown should have a `data-heap-track="navigation-header-dropdown-menu-link"` attribute on the corresponding `<a />` tag
- Visit any product page, such as [/consul][/consul], and confirm attributes appear as expected
    - The product logo link in the top nav should have a `data-heap-track="navigation-header-product-logo-link"` on the `<a />` element
    - Each link directly in the top nav should have a `data-heap-track="navigation-header-primary-nav-link"` on the `<a />` element
    - Each dropdown activator, such as `Documentation`, should have a `data-heap-track="navigation-header-dropdown-menu-activator"` attribute on the `<button />` element
    - Each item in any of the dropdowns should have a `data-heap-track="navigation-header-dropdown-menu-link"` attribute on the corresponding `<a />` tag
- And on any page on the site...
    - The Give Feedback button in the top nav should have either `data-heap-track="give-feedback-button"` (when shown in full, with text), or `data-heap-track="give-feedback-button-icon-only"` (when shown as the icon only)

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1202097197789424/1203168985343961/f
[preview]: https://dev-portal-git-zstop-nav-heap-track-hashicorp.vercel.app
[/consul]: https://dev-portal-git-zstop-nav-heap-track-hashicorp.vercel.app/consul
